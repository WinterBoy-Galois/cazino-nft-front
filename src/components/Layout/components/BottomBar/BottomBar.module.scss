@import '../../../../styles/shared.scss';

.container {
  border-top: 4px solid $light-blue;
  height: 100%;
  position: relative;
}

.balance {
  position: absolute;
  right: 0;
  top: 0;
}
