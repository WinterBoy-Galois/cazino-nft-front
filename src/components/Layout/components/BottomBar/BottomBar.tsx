import React from 'react';
import styles from './BottomBar.module.scss';
import Balance from './components/Balance';
import BottomBarMenu from './components/Menu';
import { formatBitcoin } from '../../../../common/util/format.util';

interface IProps {
  hasUnclaimedBonus?: boolean;
  balance?: number;
  onClick?: () => void;
}

const BottomBar: React.FC<IProps> = ({ hasUnclaimedBonus = true, balance = 0.0, onClick }) => {
  return (
    <div className={`container-fluid h-100`}>
      <div className={`row h-100`}>
        <div className={`col-12 ${styles.container}`}>
          <div>
            <BottomBarMenu hasUnclaimedBonus={hasUnclaimedBonus} />
          </div>
          <div className={styles.balance}>
            <Balance value={formatBitcoin(balance)} onClick={onClick} />
          </div>
        </div>
      </div>
    </div>
  );
};

export default BottomBar;
