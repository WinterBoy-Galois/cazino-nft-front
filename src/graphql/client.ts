import { ApolloClient, InMemoryCache, from, split } from '@apollo/client';
import { WebSocketLink } from '@apollo/client/link/ws';
import { TokenRefreshLink } from 'apollo-link-token-refresh';
import possibleTypes from './possibleTypes.json';
import { getMainDefinition } from '@apollo/client/utilities';
import { HttpLink, defaultDataIdFromObject } from '@apollo/client';
import jwtDecode from 'jwt-decode';
import { getEpoch } from '../common/util/date.util';
import { setContext } from '@apollo/client/link/context';
import { AuthType } from '../state/models/auth.model';
import { appConfig } from '../common/config';

const cache = new InMemoryCache({
  dataIdFromObject(responseObject) {
    switch (responseObject.__typename) {
      default:
        return defaultDataIdFromObject(responseObject);
    }
  },
  addTypename: true,
  possibleTypes,
});

let wsLink: WebSocketLink;

const getApolloClient = (
  onAccessTokenRefresh: (accessToken: string) => void,
  onSignOut: () => void,
  authType: AuthType,
  accessToken?: string
) => {
  cache.reset();

  if (wsLink) {
    (wsLink as any).subscriptionClient.close();
  }

  wsLink = new WebSocketLink({
    uri: `${appConfig.apiBasePathWS}/graphql`,
    options: {
      reconnect: true,
      connectionParams: {
        authToken: accessToken ?? undefined,
      },
    },
  });

  const httpLink = new HttpLink({
    uri: `${appConfig.apiBasePath}/graphql`,
    credentials: 'include',
  });

  const authLink = setContext((_, { headers }) => {
    if (accessToken) {
      return {
        headers: {
          ...headers,
          authorization: `Bearer ${accessToken}`,
        },
      };
    }

    return headers;
  });

  const tokenLink = new TokenRefreshLink({
    accessTokenField: 'accessToken',
    isTokenValidOrUndefined: () => {
      if (accessToken) {
        const { exp } = jwtDecode(accessToken);

        if (exp < getEpoch()) {
          return false;
        }
      } else if (authType === 'SIGNED_IN') {
        return false;
      }

      return true;
    },
    fetchAccessToken: () => {
      return fetch(`${appConfig.apiBasePath}/refresh_tokens`, {
        method: 'POST',
        credentials: 'include',
      });
    },
    handleFetch: accessToken => onAccessTokenRefresh(accessToken),
    handleError: () => onSignOut(),
  });

  const link = split(
    ({ query }) => {
      const definition = getMainDefinition(query);
      return definition.kind === 'OperationDefinition' && definition.operation === 'subscription';
    },
    wsLink,
    authLink.concat(httpLink)
  );

  return new ApolloClient({
    cache,
    link: from([tokenLink, link]),
  });
};

export default getApolloClient;
