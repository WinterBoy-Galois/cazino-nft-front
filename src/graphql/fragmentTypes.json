{
  "__schema": {
    "types": [
      {
        "kind": "UNION",
        "name": "PublicUserOrError",
        "possibleTypes": [{ "name": "PublicUser" }, { "name": "GenericErrorArray" }]
      },
      {
        "kind": "UNION",
        "name": "LoginResultOrError",
        "possibleTypes": [{ "name": "LoginResult" }, { "name": "GenericErrorArray" }]
      },
      {
        "kind": "UNION",
        "name": "UserOrError",
        "possibleTypes": [{ "name": "User" }, { "name": "GenericErrorArray" }]
      },
      {
        "kind": "UNION",
        "name": "SignedInTokenOrError",
        "possibleTypes": [{ "name": "LoginResult" }, { "name": "GenericErrorArray" }]
      },
      {
        "kind": "UNION",
        "name": "BetResultOrError",
        "possibleTypes": [{ "name": "BetResult" }, { "name": "GenericErrorArray" }]
      },
      {
        "kind": "UNION",
        "name": "BooleanOrError",
        "possibleTypes": [{ "name": "GenericBoolean" }, { "name": "GenericErrorArray" }]
      }
    ]
  }
}
