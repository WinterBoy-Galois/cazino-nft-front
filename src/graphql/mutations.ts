import gql from 'graphql-tag';

export const CHANGE_SERVER_SEED = gql`
  mutation ChangeServerSeed {
    changeServerSeed {
      ... on GenericErrorArray {
        errors {
          source
          code
          message
          args
        }
      }
    }
  }
`;

export const ACTIVATE_ACCOUNT = gql`
  mutation ActivateAccount($code: String!) {
    activateAccount(code: $code) {
      ... on GenericErrorArray {
        errors {
          source
          code
          message
          args
        }
      }
    }
  }
`;

export const SIGN_IN = gql`
  mutation SignIn($email: String, $password: String, $remember: Boolean) {
    signIn(email: $email, password: $password, remember: $remember) {
      ... on LoginResult {
        accessToken
        user {
          id
          username
          avatarUrl
          balance
          email
          isActivated
          depositAddress
          refCode
        }
      }
      ... on GenericErrorArray {
        errors {
          source
          code
          message
          args
        }
      }
    }
  }
`;

export const SIGN_UP = gql`
  mutation SignUp(
    $email: String
    $password: String
    $username: String
    $token: String
    $ref: String
  ) {
    registerUser(
      username: $username
      email: $email
      password: $password
      captcha: $token
      ref: $ref
    ) {
      ... on LoginResult {
        accessToken
        user {
          id
          username
          avatarUrl
          balance
          depositAddress
          refCode
        }
      }
      ... on GenericErrorArray {
        errors {
          source
          code
          message
          args
        }
      }
    }
  }
`;

export const SIGN_OUT = gql`
  mutation SignOut {
    signOut
  }
`;

export const RESEND_ACTIVATION_CODE = gql`
  mutation ResendActivationCode {
    resendActivationCode {
      ... on GenericBoolean {
        result
      }
      ... on GenericErrorArray {
        errors {
          source
          code
          message
          args
        }
      }
    }
  }
`;

export const RESET_PASSWORD = gql`
  mutation ResetPassword($token: String!, $newPassword: String!) {
    resetPassword(uuid: $token, newPassword: $newPassword) {
      ... on LoginResult {
        accessToken
        user {
          id
          username
          avatarUrl
          balance
          depositAddress
        }
      }
      ... on GenericErrorArray {
        errors {
          source
          code
          message
          args
        }
      }
    }
  }
`;

export const RECOVER_PASSWORD = gql`
  mutation RecoverPassword($email: String!) {
    forgotPassword(email: $email) {
      ... on GenericBoolean {
        result
      }
      ... on GenericErrorArray {
        errors {
          source
          code
          message
          args
        }
      }
    }
  }
`;

export const UPDATE_PASSWORD = gql`
  mutation UpdatePassword($oldPassword: String!, $newPassword: String!) {
    modifyPassword(oldPassword: $oldPassword, newPassword: $newPassword) {
      ... on GenericErrorArray {
        errors {
          source
          code
          message
          args
        }
      }
    }
  }
`;

export const UPDATE_PREFERENCES = gql`
  mutation UpdatePreferences($hideUsername: Boolean!, $hideProfit: Boolean!, $hideWager: Boolean!) {
    modifyPreferences(
      hideUsername: $hideUsername
      hideTotalProfit: $hideProfit
      hideTotalWager: $hideWager
    ) {
      ... on User {
        id
        hideUsername
        hideTotalProfit
        hideTotalWager
      }
      ... on GenericErrorArray {
        errors {
          source
          code
          message
          args
        }
      }
    }
  }
`;

export const UPDATE_AVATAR = gql`
  mutation updateAvatar($index: Int!) {
    modifyAvatar(index: $index) {
      ... on User {
        id
        avatarUrl
      }
      ... on GenericErrorArray {
        errors {
          source
          code
          message
          args
        }
      }
    }
  }
`;

export const MAKE_BET_DICE = gql`
  mutation makeBetDice($betAmount: Float!, $over: Boolean!, $target: Float!) {
    makeBetDice(betAmount: $betAmount, over: $over, target: $target) {
      ... on BetResultDice {
        id
        result
        lucky
        multiplier
        profit
        balance
      }
      ... on GenericErrorArray {
        errors {
          source
          code
          message
          args
        }
      }
    }
  }
`;

export const MAKE_BET_CLAMS = gql`
  mutation makeBetClams($betAmount: Float!, $selection: [Int]) {
    makeBetClams(betAmount: $betAmount, selection: $selection) {
      ... on BetResultClams {
        id
        result
        lucky
        multiplier
        profit
        balance
      }
      ... on GenericErrorArray {
        errors {
          source
          code
          message
          args
        }
      }
    }
  }
`;

export const MAKE_BET_GOALS = gql`
  mutation makeBetGoals($betAmount: Float!, $difficulty: GoalsDifficulty!) {
    makeBetGoals(betAmount: $betAmount, difficulty: $difficulty) {
      ... on GoalsGameSetup {
        session {
          betId
          betAmount
          currentStep
          selections {
            step
            luckySpots
            selected
          }
          difficulty
          profits {
            step
            multiplier
            profit
          }
          totalProfit {
            multiplier
            profit
          }
          nextProfit {
            multiplier
            profit
          }
          allowNext
          profitCut
        }
        balance
      }
      ... on GenericErrorArray {
        errors {
          source
          code
          message
          args
        }
      }
    }
  }
`;

export const ADVANCE_GOALS = gql`
  mutation advanceGoals($betId: ID!, $selection: Int!) {
    advanceGoals(betId: $betId, selection: $selection) {
      ... on GoalsStep {
        id
        totalProfit {
          multiplier
          profit
        }
        nextProfit {
          multiplier
          profit
        }
        nextStep
        profitCut
        allowNext
      }
      ... on GoalsComplete {
        id
        lucky
        profit {
          multiplier
          profit
        }
        balance
        profitCut
        result {
          step
          luckySpots
          selected
        }
      }
      ... on GenericErrorArray {
        errors {
          source
          code
          message
          args
        }
      }
    }
  }
`;

export const CASH_OUT_GOALS = gql`
  mutation cashoutGoals($betId: ID!) {
    cashoutGoals(betId: $betId) {
      ... on GoalsComplete {
        id
        lucky
        balance
        profit {
          multiplier
          profit
        }
        profitCut
        result {
          step
          luckySpots
          selected
        }
      }
      ... on GenericErrorArray {
        errors {
          source
          code
          message
          args
        }
      }
    }
  }
`;

export const WITHDRAW = gql`
  mutation withdraw($amount: Float!, $address: String!) {
    withdraw(amount: $amount, address: $address) {
      ... on GenericBoolean {
        result
      }
      ... on GenericErrorArray {
        errors {
          source
          code
          message
          args
        }
      }
    }
  }
`;

export const CLAIM_FAUCET = gql`
  mutation claimFaucet {
    claimFaucet {
      ... on User {
        balance
      }
      ... on GenericErrorArray {
        errors {
          source
          code
          message
          args
        }
      }
    }
  }
`;

export const MAKE_BET_MINES = gql`
  mutation makeBetMines($betAmount: Float!, $mines: Int!) {
    makeBetMines(betAmount: $betAmount, mines: $mines) {
      __typename
      ... on MinesGameSetup {
        session {
          ... on MinesGameSessionSetup {
            betId
            betAmount
            mines
            open
            totalProfit {
              multiplier
              profit
            }
            nextProfit {
              multiplier
              profit
            }
            allowNext
            profitCut
          }
        }
        balance
        maxProfit
      }
      ... on GenericErrorArray {
        errors {
          source
          code
          message
          args
        }
      }
    }
  }
`;

export const ADVANCE_MINES = gql`
  mutation advanceMines($betId: ID!, $selection: Int!) {
    advanceMines(betId: $betId, selection: $selection) {
      ... on MinesStep {
        id
        totalProfit {
          multiplier
          profit
        }
        nextProfit {
          multiplier
          profit
        }
        profitCut
        allowNext
        result
      }
      ... on MinesComplete {
        id
        lucky
        balance
        profit {
          multiplier
          profit
        }
        minePositions
      }
      ... on GenericErrorArray {
        errors {
          source
          code
          message
          args
        }
      }
    }
  }
`;

export const CASH_OUT_MINES = gql`
  mutation cashoutMines($betId: ID!) {
    cashoutMines(betId: $betId) {
      ... on MinesComplete {
        id
        lucky
        balance
        profit {
          multiplier
          profit
        }
        minePositions
      }
      ... on GenericErrorArray {
        errors {
          source
          code
          message
          args
        }
      }
    }
  }
`;

export const CLAIM_BONUS = gql`
  mutation claimBonus($bonusId: ID!) {
    claimBonus(bonusId: $bonusId) {
      ... on User {
        balance
      }
      ... on GenericErrorArray {
        errors {
          code
          message
        }
      }
    }
  }
`;

export const CLAIM_COMMISSION = gql`
  mutation claimCommissions {
    claimCommissions {
      ... on User {
        id
        username
        email
        balance
        hideUsername
        hideTotalProfit
        hideTotalWager
        isActivated
        avatarUrl
        totalWager
        totalProfit
        mostPlayed
        totalBets
        luckyBets
        depositAddress
        refCode
        refCommissions
      }
      ... on GenericErrorArray {
        errors {
          source
          code
          message
          args
        }
      }
    }
  }
`;
