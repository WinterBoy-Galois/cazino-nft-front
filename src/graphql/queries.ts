import gql from 'graphql-tag';

export const RECENT_BETS = gql`
  {
    recentBets {
      allBets {
        id
        time
        userid
        username
        gameid
        bet
        profit
        multiplier
      }
      myBets {
        id
        time
        userid
        username
        gameid
        bet
        profit
        multiplier
      }
    }
  }
`;

export const LEADERBOARDS = gql`
  {
    leaderboards {
      daily {
        username
        userid
        wager
        bonus
      }
      weekly {
        username
        userid
        wager
        bonus
      }
      monthly {
        username
        userid
        wager
        bonus
      }
    }
  }
`;

export const USER_INFO = gql`
  query UserInfo($userId: ID) {
    userInfo(id: $userId) {
      __typename
      ... on PublicUser {
        id
        username
        avatarUrl
        totalWager
        totalProfit
        mostPlayed
        totalBets
        luckyBets
      }
      ... on GenericErrorArray {
        errors {
          source
          code
          message
          args
        }
      }
    }
  }
`;

export const USER_INFO_AVATAR_URL = gql`
  query UserInfoAvatarUrl($userId: ID) {
    userInfo(id: $userId) {
      __typename
      ... on PublicUser {
        id
        avatarUrl
      }
      ... on GenericErrorArray {
        errors {
          source
          code
          message
          args
        }
      }
    }
  }
`;

export const BET_DETAILS = gql`
  query BetDetails($betId: ID) {
    betDetails(id: $betId) {
      ... on BetDetails {
        bet
        profit
        profitCut
        multiplier
        gameResult {
          ... on DiceResult {
            target
            over
            winChance
            resultFloat
          }
          ... on MinesResult {
            mineCount
            minePositions
            open
          }
          ... on GoalsResult {
            difficulty
            selections {
              step
              luckySpots
              selected
            }
          }
          ... on ClamsResult {
            selection
            resultInteger
          }
        }
      }
    }
  }
`;

export const BET_DETAILS_SERVER_SEED = gql`
  query BetDetailsServerSeed($betId: ID) {
    betDetails(id: $betId) {
      ... on BetDetails {
        seedDetails {
          __typename
          ... on SeedDetailsOther {
            serverSeedHash
          }
          ... on SeedDetailsOwn {
            activeGames
            serverSeedHash
            serverSeed
            clientSeed
            nonce
            results
            verificationUrl
          }
        }
      }
    }
  }
`;

export const ME = gql`
  query Me {
    me {
      id
      username
      avatarUrl
      balance
      email
      isActivated
      depositAddress
      refCommissions
      refCode
    }
  }
`;

export const AFF_STATS = gql`
  query affStats {
    affStats {
      bets
      refs
      wager
      commissions
    }
  }
`;

export const ME_STATISTICS_PREFERENCES = gql`
  query Me {
    me {
      id
      totalWager
      totalProfit
      mostPlayed
      totalBets
      luckyBets
      hideUsername
      hideTotalProfit
      hideTotalWager
    }
  }
`;

export const SETUP_CASHIER = gql`
  query SetupCashier {
    setupCashier {
      networkFee
      depositConfirmations
      minWithdraw
    }
    me {
      id
      depositAddress
    }
  }
`;

export const DEPOSITS = gql`
  query Deposits($page: Int, $limit: Int) {
    transactionsDeposit(page: $page, limit: $limit) {
      page
      limit
      total
      items {
        status
        time
        hash
        amount
      }
    }
  }
`;

export const SETUP_DICE = gql`
  query SetupDice {
    setupDice {
      ... on DiceGameSetup {
        minProbability
        maxProbability
        he
      }
    }
  }
`;

export const SETUP_CLAMS = gql`
  query setupClams {
    setupClams {
      ... on ClamsGameSetup {
        he
      }
    }
  }
`;

export const SETUP_GOAL = gql`
  query setupGoals {
    setupGoals {
      ... on GoalsGameSetup {
        session {
          betId
          betAmount
          currentStep
          selections {
            step
            luckySpots
            selected
          }
          difficulty
          profits {
            step
            multiplier
            profit
          }
          totalProfit {
            multiplier
            profit
          }
          nextProfit {
            multiplier
            profit
          }
          allowNext
          profitCut
        }
        balance
        maxProfit
      }
    }
  }
`;

export const TRANSACTION_BETS = gql`
  query Bets($page: Int, $limit: Int) {
    transactionsBet(page: $page, limit: $limit) {
      page
      limit
      total
      items {
        id
        time
        game
        amount
        multiplier
        profit
      }
    }
  }
`;

export const TRANSACTION_AFFILIATES = gql`
  query AffiliateClaims($page: Int, $limit: Int) {
    transactionsAffiliate(page: $page, limit: $limit) {
      page
      limit
      total
      items {
        id
        claimedAt
        amount
      }
    }
  }
`;

export const TRANSACTION_WITHDRAWALS = gql`
  query transactionsWithdraw($page: Int, $limit: Int) {
    transactionsWithdraw(page: $page, limit: $limit) {
      page
      limit
      total
      items {
        id
        status
        time
        hash
        address
        amount
        fee
      }
    }
  }
`;

export const TRANSACTION_BONUSES = gql`
  query transactionsBonus($page: Int, $limit: Int) {
    transactionsBonus(page: $page, limit: $limit) {
      page
      limit
      total
      items {
        id
        givenAt
        claimedAt
        expiresAt
        amount
        position
        type
        wager
      }
    }
  }
`;

export const FAUCET_INFO = gql`
  query faucetInfo {
    faucetInfo {
      ... on FaucetInfo {
        amount
        every
        canClaim
      }
      ... on GenericErrorArray {
        errors {
          source
          code
          message
          args
        }
      }
    }
  }
`;

export const SETUP_MINES = gql`
  query setupMines {
    setupMines {
      ... on MinesGameSetup {
        session {
          betId
          betAmount
          mines
          open
          totalProfit {
            multiplier
            profit
          }
          nextProfit {
            multiplier
            profit
          }
          allowNext
          profitCut
        }
        balance
        maxProfit
      }
      ... on GenericErrorArray {
        errors {
          source
          code
          message
          args
        }
      }
    }
  }
`;

export const BONUSCLAIMS = gql`
  query bonusClaims {
    bonusClaims {
      ... on BonusClaim {
        id
        givenAt
        claimedAt
        expiresAt
        amount
        position
        type
        wager
      }
    }
  }
`;
