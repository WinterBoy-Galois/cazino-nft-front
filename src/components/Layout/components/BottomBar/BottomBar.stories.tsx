import React from 'react';
import { storiesOf } from '@storybook/react';
import BottomBar from '.';

storiesOf('Components/BottomBar', module).add('default', () => <BottomBar />);
