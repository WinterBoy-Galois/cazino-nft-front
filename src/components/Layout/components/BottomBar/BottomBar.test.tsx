import React from 'react';
import { render } from '@testing-library/react';
import { MockedProvider } from '@apollo/client/testing';
import BottomBar from './BottomBar';
import { LocationProvider } from '@reach/router';
import { FAUCET_INFO } from '../../../../graphql/queries';

const mocks = [
  {
    request: {
      query: FAUCET_INFO,
    },
    result: {
      data: {
        faucetInfo: { amount: 3300331, every: 30, canClaim: true, __typename: 'FaucetInfo' },
      },
    },
  },
];

describe('BottomBar', () => {
  it('should match snapshot', () => {
    // Arrange

    // Act
    const container = render(
      <LocationProvider>
        <MockedProvider mocks={mocks} addTypename={false}>
          <BottomBar />
        </MockedProvider>
      </LocationProvider>
    );

    // Assert
    expect(container).toMatchSnapshot();
  });
});
