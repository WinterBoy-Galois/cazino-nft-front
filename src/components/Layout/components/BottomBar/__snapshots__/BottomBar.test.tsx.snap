// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomBar should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container-fluid h-100"
      >
        <div
          class="row h-100"
        >
          <div
            class="col-12 container"
          >
            <div>
              <div
                class="container"
              >
                <div
                  class="item"
                >
                  <svg
                    class="item__icon"
                    viewBox="0 0 34.45 34.81"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      data-name="Layer 2"
                      id="Layer_2"
                    >
                      <g
                        data-name="Слой 5"
                        id="Слой_5"
                      >
                        <path
                          d="M34.43 16.07l-.49-3.34a1.77 1.77 0 00-1.76-1.53h-10v6.9h10.5a1.78 1.78 0 001.75-2.03zM.5 12.73L0 16.07a1.78 1.78 0 001.76 2h10.5V11.2h-10A1.78 1.78 0 00.5 12.73zM31.94 33V19.47h-9.77v15.34h7.9A1.87 1.87 0 0031.94 33zM2.51 19.47V33a1.86 1.86 0 001.86 1.86h7.91V19.47z"
                        />
                        <rect
                          height="23.61"
                          rx="1.29"
                          width="7.19"
                          x="13.63"
                          y="11.2"
                        />
                        <path
                          d="M26.47 5.66c-1.75-2.51-5.26.26-7 2.19a4 4 0 00-.67-.49C22.28 3.23 29.05-3.51 33 2.2c2.43 3.49-.38 7.4-11.67 8.11a3.76 3.76 0 00-.1-.66c4.77-.51 6.38-2.35 5.24-3.99z"
                        />
                        <path
                          d="M13 10.31a4.19 4.19 0 018.34 0z"
                        />
                        <path
                          d="M8 5.66c1.75-2.51 5.26.26 7 2.19a4 4 0 01.67-.49C12.16 3.23 5.38-3.51 1.4 2.2c-2.4 3.49.38 7.4 11.67 8.11a3.76 3.76 0 01.1-.66C8.45 9.14 6.83 7.3 8 5.66z"
                        />
                      </g>
                    </g>
                  </svg>
                  <div
                    class="item__badge"
                  >
                    <div
                      class="badge"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="balance"
            >
              <div
                class="wrapper"
              >
                <div
                  class="container"
                >
                  <div
                    class="border"
                  />
                  <div
                    class="value"
                  >
                    <svg
                      class="icon"
                      viewBox="0 0 64 64"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <g
                        transform="translate(0.00630876,-0.00301984)"
                      >
                        <path
                          d="m63.033,39.744c-4.274,17.143-21.637,27.576-38.782,23.301-17.138-4.274-27.571-21.638-23.295-38.78,4.272-17.145,21.635-27.579,38.775-23.305,17.144,4.274,27.576,21.64,23.302,38.784z"
                        />
                        <path
                          class="icon__inner"
                          d="m46.103,27.444c0.637-4.258-2.605-6.547-7.038-8.074l1.438-5.768-3.511-0.875-1.4,5.616c-0.923-0.23-1.871-0.447-2.813-0.662l1.41-5.653-3.509-0.875-1.439,5.766c-0.764-0.174-1.514-0.346-2.242-0.527l0.004-0.018-4.842-1.209-0.934,3.75s2.605,0.597,2.55,0.634c1.422,0.355,1.679,1.296,1.636,2.042l-1.638,6.571c0.098,0.025,0.225,0.061,0.365,0.117-0.117-0.029-0.242-0.061-0.371-0.092l-2.296,9.205c-0.174,0.432-0.615,1.08-1.609,0.834,0.035,0.051-2.552-0.637-2.552-0.637l-1.743,4.019,4.569,1.139c0.85,0.213,1.683,0.436,2.503,0.646l-1.453,5.834,3.507,0.875,1.439-5.772c0.958,0.26,1.888,0.5,2.798,0.726l-1.434,5.745,3.511,0.875,1.453-5.823c5.987,1.133,10.489,0.676,12.384-4.739,1.527-4.36-0.076-6.875-3.226-8.515,2.294-0.529,4.022-2.038,4.483-5.155zm-8.022,11.249c-1.085,4.36-8.426,2.003-10.806,1.412l1.928-7.729c2.38,0.594,10.012,1.77,8.878,6.317zm1.086-11.312c-0.99,3.966-7.1,1.951-9.082,1.457l1.748-7.01c1.982,0.494,8.365,1.416,7.334,5.553z"
                        />
                      </g>
                    </svg>
                    <span>
                      0.00000000
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container-fluid h-100"
    >
      <div
        class="row h-100"
      >
        <div
          class="col-12 container"
        >
          <div>
            <div
              class="container"
            >
              <div
                class="item"
              >
                <svg
                  class="item__icon"
                  viewBox="0 0 34.45 34.81"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    data-name="Layer 2"
                    id="Layer_2"
                  >
                    <g
                      data-name="Слой 5"
                      id="Слой_5"
                    >
                      <path
                        d="M34.43 16.07l-.49-3.34a1.77 1.77 0 00-1.76-1.53h-10v6.9h10.5a1.78 1.78 0 001.75-2.03zM.5 12.73L0 16.07a1.78 1.78 0 001.76 2h10.5V11.2h-10A1.78 1.78 0 00.5 12.73zM31.94 33V19.47h-9.77v15.34h7.9A1.87 1.87 0 0031.94 33zM2.51 19.47V33a1.86 1.86 0 001.86 1.86h7.91V19.47z"
                      />
                      <rect
                        height="23.61"
                        rx="1.29"
                        width="7.19"
                        x="13.63"
                        y="11.2"
                      />
                      <path
                        d="M26.47 5.66c-1.75-2.51-5.26.26-7 2.19a4 4 0 00-.67-.49C22.28 3.23 29.05-3.51 33 2.2c2.43 3.49-.38 7.4-11.67 8.11a3.76 3.76 0 00-.1-.66c4.77-.51 6.38-2.35 5.24-3.99z"
                      />
                      <path
                        d="M13 10.31a4.19 4.19 0 018.34 0z"
                      />
                      <path
                        d="M8 5.66c1.75-2.51 5.26.26 7 2.19a4 4 0 01.67-.49C12.16 3.23 5.38-3.51 1.4 2.2c-2.4 3.49.38 7.4 11.67 8.11a3.76 3.76 0 01.1-.66C8.45 9.14 6.83 7.3 8 5.66z"
                      />
                    </g>
                  </g>
                </svg>
                <div
                  class="item__badge"
                >
                  <div
                    class="badge"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="balance"
          >
            <div
              class="wrapper"
            >
              <div
                class="container"
              >
                <div
                  class="border"
                />
                <div
                  class="value"
                >
                  <svg
                    class="icon"
                    viewBox="0 0 64 64"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      transform="translate(0.00630876,-0.00301984)"
                    >
                      <path
                        d="m63.033,39.744c-4.274,17.143-21.637,27.576-38.782,23.301-17.138-4.274-27.571-21.638-23.295-38.78,4.272-17.145,21.635-27.579,38.775-23.305,17.144,4.274,27.576,21.64,23.302,38.784z"
                      />
                      <path
                        class="icon__inner"
                        d="m46.103,27.444c0.637-4.258-2.605-6.547-7.038-8.074l1.438-5.768-3.511-0.875-1.4,5.616c-0.923-0.23-1.871-0.447-2.813-0.662l1.41-5.653-3.509-0.875-1.439,5.766c-0.764-0.174-1.514-0.346-2.242-0.527l0.004-0.018-4.842-1.209-0.934,3.75s2.605,0.597,2.55,0.634c1.422,0.355,1.679,1.296,1.636,2.042l-1.638,6.571c0.098,0.025,0.225,0.061,0.365,0.117-0.117-0.029-0.242-0.061-0.371-0.092l-2.296,9.205c-0.174,0.432-0.615,1.08-1.609,0.834,0.035,0.051-2.552-0.637-2.552-0.637l-1.743,4.019,4.569,1.139c0.85,0.213,1.683,0.436,2.503,0.646l-1.453,5.834,3.507,0.875,1.439-5.772c0.958,0.26,1.888,0.5,2.798,0.726l-1.434,5.745,3.511,0.875,1.453-5.823c5.987,1.133,10.489,0.676,12.384-4.739,1.527-4.36-0.076-6.875-3.226-8.515,2.294-0.529,4.022-2.038,4.483-5.155zm-8.022,11.249c-1.085,4.36-8.426,2.003-10.806,1.412l1.928-7.729c2.38,0.594,10.012,1.77,8.878,6.317zm1.086-11.312c-0.99,3.966-7.1,1.951-9.082,1.457l1.748-7.01c1.982,0.494,8.365,1.416,7.334,5.553z"
                      />
                    </g>
                  </svg>
                  <span>
                    0.00000000
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
